src/org/labkey/response/ResponseUpgradeCode.java

```diff
@@ -0,0 +1,130 @@
+package org.labkey.response;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.labkey.api.data.Container;
+import org.labkey.api.data.ContainerManager;
+import org.labkey.api.data.DeferredUpgrade;
+import org.labkey.api.data.PropertyManager;
+import org.labkey.api.data.UpgradeCode;
+import org.labkey.api.module.Module;
+import org.labkey.api.module.ModuleContext;
+import org.labkey.api.module.ModuleLoader;
+import org.labkey.api.module.ModuleProperty;
+import org.labkey.api.util.StringUtilsLabKey;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static java.util.Base64.getDecoder;
+import static org.labkey.response.ResponseController.ServerConfigurationAction.RESPONSE_SERVER_CONFIGURATION;
+import static org.labkey.response.ResponseController.ServerConfigurationAction.WCP_SERVER;
+
+public class ResponseUpgradeCode implements UpgradeCode
+{
+    private static final Logger LOG = LogManager.getLogger(ResponseUpgradeCode.class);
+    private static final String SURVEY_METADATA_DIRECTORY = "SurveyMetadataDirectory";
+    private static final String METADATA_SERVICE_BASE_URL = "MetadataServiceBaseUrl";
+    private static final String METADATA_SERVICE_ACCESS_TOKEN = "MetadataServiceAccessToken";
+
+    /**
+     * Invoked within mobileappstudy-21.000-21.001.sql
+     * Response Module Properties, which were formerly set in the Folder Management Module Properties tab, are here migrated
+     * to the Admin Console Response Server Configuration page.
+     */
+    @DeferredUpgrade
+    public static void migrateResponseServerConfig(ModuleContext context)
+    {
+        Module responseModule = ModuleLoader.getInstance().getModule("Response");
+        if (responseModule == null)
+            return;
+
+        String surveyMetadataDirectoryValue = getResultingPropertyValue(SURVEY_METADATA_DIRECTORY, responseModule);
+        String metadataServiceBaseUrl = getResultingPropertyValue(METADATA_SERVICE_BASE_URL, responseModule);
+        String metadataServiceAccessToken =  getResultingPropertyValue(METADATA_SERVICE_ACCESS_TOKEN, responseModule);
+
+        PropertyManager.PropertyMap props = PropertyManager.getEncryptedStore().getWritableProperties(ContainerManager.getRoot(), RESPONSE_SERVER_CONFIGURATION, false);
```

Comment:
```suggestion
        PropertyManager.PropertyMap props = PropertyManager.getEncryptedStore().getWritableProperties(ContainerManager.getRoot(), RESPONSE_SERVER_CONFIGURATION, true);
```
Very important to create the property map, otherwise props will be `null` and you'll never save the migrated properties

Reply:
AGH didn't catch this because it was already spun up, obviously, on my machine while I was testing the upgrade script. Change made.